### PR TITLE
OC-11966:Update Casebook PDF header and footer

### PR DESCRIPTION
**Pull request by Taoli2018**

### CODE REVIEW

File: src/main/java/org/akaza/openclinica/controller/StudyParticipantController.java [hunk 1]
```diff
@@ -646,6 +670,7 @@ private String startBulkCaseBookPDFJob(String schema,
 			String accessToken = (String) request.getSession().getAttribute("accessToken");
 			servletContext.setAttribute("accessToken", accessToken);
 			servletContext.setAttribute("studyID", study.getStudyId()+"");
+			servletContext.setAttribute("pdfHeader", pdfHeader);
```
Comment (svadla-oc) [inline src/main/java/org/akaza/openclinica/controller/StudyParticipantController.java]: Why set this as a servlet context attribute and not pass it in as a method argument?

Comment (svadla-oc) [reply src/main/java/org/akaza/openclinica/controller/StudyParticipantController.java]: Also, can pdf header be calculated within the Future?

File: src/main/java/org/akaza/openclinica/controller/StudyParticipantController.java [hunk 2]
```diff
@@ -610,13 +610,25 @@ private String startBulkCaseBookPDFJob(String schema,
 										   String landscape) {
 									 	 
 
+		    String siteName = null;
+		    String studyName = null;
+		    String pdfHeader =  "StudyName: SiteName - Participant ParticipantID";
```
Comment (svadla-oc) [inline src/main/java/org/akaza/openclinica/controller/StudyParticipantController.java]: Instead of using string replacement, use MessageFormat https://docs.oracle.com/javase/8/docs/api/java/text/MessageFormat.html

File: src/main/java/core/org/akaza/openclinica/dao/hibernate/AbstractDomainDao.java [hunk 3]
```diff
@@ -50,6 +50,9 @@ public String getDomainClassName() {
 
     @SuppressWarnings("unchecked")@Transactional
     public T findByOcOID(String OCOID) {
+    	if(OCOID == null) {
+    		return null;
+    	}
```
Comment (svadla-oc) [inline src/main/java/core/org/akaza/openclinica/dao/hibernate/AbstractDomainDao.java]: why did you add this?

File: src/main/java/core/org/akaza/openclinica/service/StudyParticipantServiceImpl.java [hunk 4]
```diff
@@ -45,6 +45,7 @@
 import java.io.File;
 import java.io.IOException;
 import java.io.PrintWriter;
+import java.text.MessageFormat;
```
Comment (svadla-oc) [inline src/main/java/core/org/akaza/openclinica/service/StudyParticipantServiceImpl.java]: Delete this import as you are not using it in this class

File: src/main/java/org/akaza/openclinica/controller/StudyParticipantController.java (hunk 1 above)
Comment (Taoli2018) [reply src/main/java/org/akaza/openclinica/controller/StudyParticipantController.java]: changed --- after discuss, we decided to pass in Study object as method parameter

File: src/main/java/org/akaza/openclinica/controller/StudyParticipantController.java (hunk 2 above)
Comment (Taoli2018) [reply src/main/java/org/akaza/openclinica/controller/StudyParticipantController.java]: done

File: src/main/java/org/akaza/openclinica/service/PdfServiceImpl.java [hunk 5]
```diff
@@ -181,7 +182,32 @@ public int addHeaderOrFooter(PDDocument document, String headerMsg,String footer
...
 
 
-
+   public String preparePdfHeader(Study study, Study site, String studySubjectIdentifier) {
+	  
+	    String siteName = null;
+	    String studyName = null;
+	    String participantID = studySubjectIdentifier.trim();
+	    		   
+	    MessageFormat pdfHeaderFormat1 =  new MessageFormat("{0}: {1} - Participant {2}");
```
Comment (svadla-oc) [inline src/main/java/org/akaza/openclinica/service/PdfServiceImpl.java]: Make these variables static final, so you don't have to initialize them every time

File: src/main/java/org/akaza/openclinica/service/PdfServiceImpl.java [hunk 6]
```diff
@@ -181,7 +182,32 @@ public int addHeaderOrFooter(PDDocument document, String headerMsg,String footer
...
+	    MessageFormat pdfHeaderFormat2 =  new MessageFormat("{0} - Participant {2}");
```
Comment (svadla-oc) [inline src/main/java/org/akaza/openclinica/service/PdfServiceImpl.java]: Instead of {2}, this should be {1}

File: src/main/java/org/akaza/openclinica/controller/StudyParticipantController.java [hunk 7]
```diff
@@ -610,54 +610,24 @@ private String startBulkCaseBookPDFJob(String schema,
...
-				study = studyDao.findByOcOID(studyOid.trim());
-				studyName = study.getName();
-				
-				if(study.getStudy() != null) {
-					site = study;
-					study = study.getStudy(); 
-				}
-
-			}
+		    final	Study 	site = studyDao.findByOcOID(siteOid);
```
Comment (svadla-oc) [inline src/main/java/org/akaza/openclinica/controller/StudyParticipantController.java]: change to: final Study site = siteOid == null ? null : studyDao.findByOcOID(siteOid)

File: src/main/java/core/org/akaza/openclinica/dao/hibernate/AbstractDomainDao.java (hunk 3 above)
Comment (Taoli2018) [reply src/main/java/core/org/akaza/openclinica/dao/hibernate/AbstractDomainDao.java]: removed, changed at local

File: src/main/java/org/akaza/openclinica/service/PdfServiceImpl.java (hunk 5 above)
Comment (Taoli2018) [reply src/main/java/org/akaza/openclinica/service/PdfServiceImpl.java]: done

File: src/main/java/org/akaza/openclinica/controller/StudyParticipantController.java (hunk 7 above)
Comment (Taoli2018) [reply src/main/java/org/akaza/openclinica/controller/StudyParticipantController.java]: done

File: src/main/java/core/org/akaza/openclinica/service/StudyParticipantServiceImpl.java (hunk 4 above)
Comment (Taoli2018) [reply src/main/java/core/org/akaza/openclinica/service/StudyParticipantServiceImpl.java]: done

**Comment by svadla-oc** [comment]
👍
